openmc/data/thermal.py

```diff
@@ -582,3 +586,47 @@ def from_ace(cls, ace_or_filename, name=None):
                                 table.nuclides.append(isotope)
 
         return table
+
+    @classmethod
+    def from_njoy(cls, filename, filename_thermal, temperatures=None, **kwargs):
+        """Generate incident neutron data by running NJOY.
+
+        Parameters
+        ----------
+        filename : str
+            Path to ENDF neutron sublibrary file
+        filename_thermal : str
+            Path to ENDF thermal scattering sublibrary file
+        temperatures : iterable of float
+            Temperatures in Kelvin to produce data at. If omitted, data is
+            produced at all temperatures in the ENDF thermal scattering
+            sublibrary.
+        **kwargs
+            Keyword arguments passed to :func:`openmc.data.njoy.make_ace_thermal`
+
+        Returns
+        -------
+        data : openmc.data.ThermalScattering
+            Thermal scattering data
+
+        """
+        # Create temporary directory -- it would be preferable to use
+        # TemporaryDirectory(), but it is only available in Python 3.2
```

Comment:
Yes, by the time Python 2 dies (2020) or before, I would advocate switching over to a Py3-only codebase.